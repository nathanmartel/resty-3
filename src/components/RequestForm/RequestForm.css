input[type=text], textarea, label {
  width: 100%;
  display: block;
}

::placeholder { color: #AAA; }

.radioControls {
  margin: 1rem 0;
  label {
    width: auto;
    display: inline-block;
  }
  label>input[type=radio] {
    margin-left: 20px;
    margin-right: 10px;
  }
  label:first-of-type>input[type=radio] {
    margin-left: 0px;
  }
}

.authHeaders {
  background-color: #f6f6f6;
  padding: 1rem;
  margin-bottom: 1.5rem;
  div { margin-bottom: .5rem; }
  select { display: block; }
  legend {
    font-size: 1em;
    margin-bottom: 0;
  }
  input[type=text] {
    width: 50%;
    display: inline-block;
  }
  .bearerToken input[type=text] {
    width: 100%;
  }

}

