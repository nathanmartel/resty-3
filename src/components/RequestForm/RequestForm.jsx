import React from 'react';
import PropTypes from 'prop-types';
import styles from './RequestForm.css';

const RequestForm = ({ 
  url, 
  method, 
  body, 
  onUrlChange, 
  onMethodChange, 
  onBodyChange, 
  authType, 
  authUsername, 
  authPassword, 
  authToken, 
  authUsernamePlaceholder, 
  authPasswordPlaceholder, 
  authTokenPlaceholder, 
  onAuthTypeChange, 
  onAuthUsernameChange, 
  onAuthPasswordChange, 
  onAuthTokenChange, 
  onSubmit }) => { 

  return (
    <div className="requestContainer">
      <h3>Request</h3>
      <form onSubmit={onSubmit}>
        <label>URL:
          <input type="text" value={url} onChange={onUrlChange} />
        </label>

        <div className={styles.radioControls}>
          <label>
            <input type="radio" name="method" value="GET" checked={method === 'GET'} onChange={onMethodChange}/>
            GET
          </label>
          <label>
            <input type="radio" name="method" value="POST" checked={method === 'POST'} onChange={onMethodChange} />
            POST
          </label>
          <label>
            <input type="radio" name="method" value="PATCH" checked={method === 'PATCH'} onChange={onMethodChange} />
            PATCH
          </label>
          <label>
            <input type="radio" name="method" value="PUT" checked={method === 'PUT'} onChange={onMethodChange} />
            PUT
          </label>
          <label>
            <input type="radio" name="method" value="DELETE" checked={method === 'DELETE'} onChange={onMethodChange} />
            DELETE
          </label>
        </div>

        <label>Body:
          <textarea rows="5" value={body} onChange={onBodyChange} />
        </label>
        <br />

        <div className={styles.authHeaders}>
          <h4>Headers</h4>
          <div className={styles.authSelect}>
            <label>Authorization Type:
              <select onChange={onAuthTypeChange} value={authType}>               
                <option value="">None</option>
                <option value="basic">Basic Authorization</option>
                <option value="bearerToken">Bearer Token</option>
              </select>
            </label>
          </div>

          <div className={styles.basicAuth}>
            <fieldset>
              <legend>Basic Authorization:</legend>
              <input type="text" placeholder={authUsernamePlaceholder} value={authUsername} onChange={onAuthUsernameChange} />
              <input type="text" placeholder={authPasswordPlaceholder} value={authPassword} onChange={onAuthPasswordChange} />
            </fieldset>
          </div>

          <div className={styles.bearerToken}>
            <label>Bearer Token:
              <input type="text" placeholder={authTokenPlaceholder} value={authToken} onChange={onAuthTokenChange} />
            </label>
          </div>

        </div>


        <button type="submit">Submit</button>
      </form>
    </div>
  );
};

RequestForm.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  onUrlChange: PropTypes.func.isRequired,
  onMethodChange: PropTypes.func.isRequired,
  onBodyChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  authType: PropTypes.string.isRequired,
  authUsername: PropTypes.string.isRequired,
  authPassword: PropTypes.string.isRequired,
  authToken: PropTypes.string.isRequired,
  authUsernamePlaceholder: PropTypes.string.isRequired,
  authPasswordPlaceholder: PropTypes.string.isRequired,
  authTokenPlaceholder: PropTypes.string.isRequired,
  onAuthTypeChange: PropTypes.func.isRequired, 
  onAuthUsernameChange: PropTypes.func.isRequired,
  onAuthPasswordChange: PropTypes.func.isRequired,
  onAuthTokenChange: PropTypes.func.isRequired
};

export default RequestForm;
