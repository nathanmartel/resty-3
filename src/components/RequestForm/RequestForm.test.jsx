import React from 'react';
import { shallow } from 'enzyme';
import RequestForm from './RequestForm';

describe('RequestForm component', () => {
  it('renders RequestForm', () => {
    const wrapper = shallow(<RequestForm
      url={'www.test.com'} 
      method={'GET'} 
      body={'{ test: "test" }'}
      onUrlChange={() => {}}
      onMethodChange={() => {}}
      onBodyChange={() => {}}
      authType={'none'}
      authUsername={''}
      authPassword={''}
      authToken={''}
      authUsernamePlaceholder={'Username'}
      authPasswordPlaceholder={'Password'}
      authTokenPlaceholder={'Token'}
      onAuthTypeChange={() => {}}
      onAuthUsernameChange={() => {}}
      onAuthPasswordChange={() => {}}
      onAuthTokenChange={() => {}}
      onSubmit={() => {}}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
