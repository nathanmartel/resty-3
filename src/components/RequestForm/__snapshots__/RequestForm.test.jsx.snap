// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestForm component renders RequestForm 1`] = `
<div
  className="requestContainer"
>
  <h3>
    Request
  </h3>
  <form
    onSubmit={[Function]}
  >
    <label>
      URL:
      <input
        onChange={[Function]}
        type="text"
        value="www.test.com"
      />
    </label>
    <div
      className="radioControls"
    >
      <label>
        <input
          checked={true}
          name="method"
          onChange={[Function]}
          type="radio"
          value="GET"
        />
        GET
      </label>
      <label>
        <input
          checked={false}
          name="method"
          onChange={[Function]}
          type="radio"
          value="POST"
        />
        POST
      </label>
      <label>
        <input
          checked={false}
          name="method"
          onChange={[Function]}
          type="radio"
          value="PATCH"
        />
        PATCH
      </label>
      <label>
        <input
          checked={false}
          name="method"
          onChange={[Function]}
          type="radio"
          value="PUT"
        />
        PUT
      </label>
      <label>
        <input
          checked={false}
          name="method"
          onChange={[Function]}
          type="radio"
          value="DELETE"
        />
        DELETE
      </label>
    </div>
    <label>
      Body:
      <textarea
        onChange={[Function]}
        rows="5"
        value="{ test: \\"test\\" }"
      />
    </label>
    <br />
    <div
      className="authHeaders"
    >
      <h4>
        Headers
      </h4>
      <div
        className="authSelect"
      >
        <label>
          Authorization Type:
          <select
            onChange={[Function]}
            value="none"
          >
            <option
              value=""
            >
              None
            </option>
            <option
              value="basic"
            >
              Basic Authorization
            </option>
            <option
              value="bearerToken"
            >
              Bearer Token
            </option>
          </select>
        </label>
      </div>
      <div
        className="basicAuth"
      >
        <fieldset>
          <legend>
            Basic Authorization:
          </legend>
          <input
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value=""
          />
          <input
            onChange={[Function]}
            placeholder="Password"
            type="text"
            value=""
          />
        </fieldset>
      </div>
      <div
        className="bearerToken"
      >
        <label>
          Bearer Token:
          <input
            onChange={[Function]}
            placeholder="Token"
            type="text"
            value=""
          />
        </label>
      </div>
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
